core/src/test/scala/org/apache/spark/scheduler/ReplayListenerSuite.scala

```diff
@@ -236,20 +238,15 @@ class ReplayListenerSuite extends SparkFunSuite with BeforeAndAfter with LocalSp
   /**
    * A simple listener that buffers all the events it receives.
    *
-   * The event buffering functionality must be implemented within EventLoggingListener itself.
-   * This is because of the following race condition: the event may be mutated between being
-   * processed by one listener and being processed by another. Thus, in order to establish
-   * a fair comparison between the original events and the replayed events, both functionalities
-   * must be implemented within one listener (i.e. the EventLoggingListener).
-   *
-   * This child listener inherits only the event buffering functionality, but does not actually
-   * log the events.
    */
-  private class EventMonster(conf: SparkConf)
-    extends EventLoggingListener("test", None, new URI("testdir"), conf) {
+  private class EventMonster extends SparkFirehoseListener {
```

Comment:
No need to change it here, but, Event_Monster_?

Reply:
lol, good point.  EventBufferingListener would be better

Reply:
Sure, I will rename it EventBufferingListener.

Reply:
Updated the PR description as well to not mention EventMonster.